### ISSUE TITLE
Build and deploy a FDU on a node? 

### ISSUE THREAD

**Issue by zrnapoleon** [eclipse/fog05#146]
For now, I have successfully installed the fog with its 5 plugins. But when i tried to deploy a LXD FDU:
$ python3 lifecycle_non_lxd_test.py 127.0.0.1:7887 fdu_lxd_test.json
Here  lifecycle_non_lxd_test.py and  fdu_lxd_test.json are based on:
https://github.com/atolab/fog05_demo/blob/master/fim_api/lifecycle_nonetwork.py
and
https://github.com/atolab/fog05_demo/blob/master/fim_api/fdu_lxd.json 
with mu node ID:
![image](https://user-images.githubusercontent.com/53283508/64096346-fa9d2500-cd93-11e9-998f-416ed20d79a3.png)
and the net interface:
![image](https://user-images.githubusercontent.com/53283508/64096682-d726aa00-cd94-11e9-9f8f-3d0e1cf2b35e.png)
I got the following error:
![image](https://user-images.githubusercontent.com/53283508/64096996-a430e600-cd95-11e9-9dc4-bb4deef64f2c.png)

[Json and Python fiches.zip](https://github.com/eclipse/fog05/files/3564896/Json.and.Python.fiches.zip)
Hope you can give me some help, thanks!

**Comment by gabrik** [comment]
Hi @zrnapoleon, can you please paste the output of the LXD plugin?

This seems related to some permission issue to the user that is running the LXD plugin, to I guess we can get more information from that log

**Comment by zrnapoleon** [comment]
Hi @gabrik , the output of LXD plugin:
![image](https://user-images.githubusercontent.com/53283508/64100226-bd3d9500-cd9d-11e9-8751-ad755d3260f5.png)

**Comment by gabrik** [comment]
I see, this is some strange permisison error coming from LXD APIs.

Try to run this:

```
$ sudo -u fos python3
> from pylxd import Client
> c = Client()
> c.profiles.create('test') 
```
You should get the same error, can you confirm?

**Comment by zrnapoleon** [comment]
I did, but got a conection error when I rename the c = Client() before passing the test:

[Error.txt](https://github.com/eclipse/fog05/files/3565157/Error.txt)

**Comment by gabrik** [comment]
Ok, I still think LXD as some strange configuration.

Try this:
```
$ sudo -u fos python3
> from pylxd import Client
> c = Client(endpoint='https://127.0.0.1:8443', verify=False)
> c.profiles.create('test') 
```

And you should get the same unauthorized error

**Comment by zrnapoleon** [comment]
![image](https://user-images.githubusercontent.com/53283508/64101640-ddbb1e80-cda0-11e9-86f8-2c4e61985780.png)
Yeah, it is, It return the same error : nor authorized

**Comment by gabrik** [comment]
Ok, can you paste the output of:
```
$ sudo -u fos groups
```

On our test machine it returns `fos sudo lxd`, you should get the same

**Comment by zrnapoleon** [comment]
Yeah , it is:
ubuntu@chenyanjun-3:~/fog05_demo/fim_api$ sudo -u fos groups
fos sudo lxd

**Comment by gabrik** [comment]
After some investigation, this can be related to https://github.com/lxc/pylxd/issues/365, so you can try to add this environment variable in a system-wide manner:
```
echo "export LXD_DIR=/var/snap/lxd/common/lxd" >> /etc/environment/
```

**Comment by gabrik** [comment]
Fixed by #147
